repo-client/src/main/kotlin/io/lionweb/repoclient/LionWebClient.kt

```diff
@@ -0,0 +1,734 @@
+        if (node.children.isNotEmpty()) {
+            throw IllegalArgumentException("When creating a partition, please specify a single node")
+        }
+        treeStoringOperation(node, "createPartitions")
+    }
+
+    fun deletePartition(node: Node) {
+        deletePartition(node.id ?: throw IllegalStateException("Node ID not specified"))
+    }
+
+    fun deletePartition(nodeID: String) {
+        val body: RequestBody = "[\"$nodeID\"]".toRequestBody(JSON)
+        val request: Request =
+            Request.Builder()
+                .url("http://$hostname:$port/bulk/deletePartitions")
+                .considerAuthenticationToken()
+                .post(body)
+                .build()
+        httpClient.newCall(request).execute().use { response ->
+            if (response.code != HttpURLConnection.HTTP_OK) {
+                val body = response.body?.string()
+                if (debug) {
+                    println("  Response: ${response.code}")
+                    println("  Response: $body")
+                }
+                throw RuntimeException("Request failed with code ${response.code}: $body")
+            }
+        }
+    }
+
+    private fun Request.Builder.considerAuthenticationToken(): Request.Builder {
+        return if (authorizationToken == null) {
+            this
+        } else {
+            this.addHeader("Authorization", authorizationToken)
+        }
+    }
+
+    fun getPartitionIDs(): List<String> {
+        val url = "http://$hostname:$port/bulk/listPartitions"
+        val request: Request =
+            Request.Builder()
+                .url(url)
+                .considerAuthenticationToken()
+                .addHeader("Accept-Encoding", "gzip")
+                .post(EMPTY_REQUEST)
+                .build()
+        httpClient.newCall(request).execute().use { response ->
+            if (response.code == HttpURLConnection.HTTP_OK) {
+                val data =
+                    (response.body ?: throw IllegalStateException("Response without body when querying $url")).string()
+                return processChunkResponse(data) {
+                    val chunk = LowLevelJsonSerialization().deserializeSerializationBlock(it)
+                    chunk.classifierInstances.mapNotNull { it.id }
+                }
+            } else {
+                throw RuntimeException("Got back ${response.code}: ${response.body?.string()}")
+            }
+        }
+    }
+
+    // Nodes
+
+    fun retrieve(
+        rootId: String,
+        withProxyParent: Boolean = false,
+        retrievalMode: RetrievalMode = RetrievalMode.ENTIRE_SUBTREE,
+    ): Node {
+        require(rootId.isNotBlank())
+        val body: RequestBody = "{\"ids\":[\"$rootId\"] }".toRequestBody(JSON)
+        val url = "http://$hostname:$port/bulk/retrieve"
+        val urlBuilder = url.toHttpUrlOrNull()!!.newBuilder()
+        val limit =
+            when (retrievalMode) {
+                RetrievalMode.ENTIRE_SUBTREE -> "99"
```

Comment:
Maybe use the value of the JavaScript Number.MAX_SAFE_INTEGER here? 99 is rather low.

Reply:
Right, I created a constant called `MAX_LEVELs` that is defined as `Int.MAX_VALUE`

Reply:
Just make sure it isn't bigger than the JavaScript  Number.MAX_SAFE_INTEGER , because otherwise the server might not be able to convert this into a number.